com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/StrongReferencerCacheAdapter.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+package com.avaloq.tools.ddk.xtext.resource;
+
+import com.avaloq.tools.ddk.caching.CacheConfiguration;
+
+
+/**
+ * A generic resource set cache adapter which uses strong references.
+ *
+ * @param <V>
+ *          the type of the stored values
+ */
+public class StrongReferencerCacheAdapter<V> extends CacheAdapter<V> {

```

Comment:
are you sure we actually need this class? Could we perhaps just pass a boolean argument to CacheAdapter to use strong references instead of soft ones? Then the consumer can decide which references he wants without inheriting from a different class